examples/tv-casting-app/android/App/app/src/main/jni/cpp/core/MatterCastingPlayer-JNI.cpp

```diff
@@ -165,6 +165,32 @@ JNI_METHOD(void, disconnect)
     castingPlayer->Disconnect();
 }
 
+JNI_METHOD(jstring, getConnectionStateNative)
+(JNIEnv * env, jobject thiz)
+{
+    chip::DeviceLayer::StackLock lock;
+    ChipLogProgress(AppServer, "MatterCastingPlayer-JNI::getConnectionState()");
+
+    CastingPlayer * castingPlayer = support::convertCastingPlayerFromJavaToCpp(thiz);
+    VerifyOrReturnValue(castingPlayer != nullptr, env->NewStringUTF("Cast Player is nullptr"));
+
+    matter::casting::core::ConnectionState state = castingPlayer->GetConnectionState();
+    switch (state)
+    {
+    case matter::casting::core::ConnectionState::CASTING_PLAYER_NOT_CONNECTED:
+        return env->NewStringUTF("NOT_CONNECTED");
```

Comment:
please try to use CharToStringUTF, context is from
https://github.com/project-chip/connectedhomeip/blob/1612d529cd3cb4e1416f36d4dcf47c9ab57873cd/src/lib/support/JniReferences.h#L237